x-pack/plugins/observability_solution/slo/public/embeddable/slo/alerts/slo_alerts_embeddable_factory.tsx

```diff
@@ -0,0 +1,133 @@
+import { SloAlertsWrapper } from './slo_alerts_wrapper';
+const history = createBrowserHistory();
+const queryClient = new QueryClient();
+
+export const getAlertsPanelTitle = () =>
+  i18n.translate('xpack.slo.sloAlertsEmbeddable.displayTitle', {
+    defaultMessage: 'SLO Alerts',
+  });
+
+export function getAlertsEmbeddableFactory(deps: SloEmbeddableDeps, kibanaVersion: string) {
+  const factory: ReactEmbeddableFactory<SloAlertsEmbeddableState, SloAlertsApi> = {
+    type: SLO_ALERTS_EMBEDDABLE_ID,
+    deserializeState: (state) => {
+      return state.rawState as SloAlertsEmbeddableState;
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+      const defaultTitle$ = new BehaviorSubject<string | undefined>(getAlertsPanelTitle());
+      const slos$ = new BehaviorSubject(state.slos);
+      const showAllGroupByInstances$ = new BehaviorSubject(state.showAllGroupByInstances);
+      const reload$ = new Subject<FetchContext>();
+      const api = buildApi(
+        {
+          ...titlesApi,
+          defaultPanelTitle: defaultTitle$,
+          serializeState: () => {
+            return {
+              rawState: {
+                ...serializeTitles(),
+                slos: slos$.getValue(),
+                showAllGroupByInstances: showAllGroupByInstances$.getValue(),
+              },
+            };
+          },
+          getSloAlertsConfig: () => {
+            return {
+              slos: slos$.getValue(),
+              showAllGroupByInstances: showAllGroupByInstances$.getValue(),
+            };
+          },
+          updateSloAlertsConfig: (update) => {
+            slos$.next(update.slos);
+            showAllGroupByInstances$.next(update.showAllGroupByInstances);
+          },
+        },
+        {
+          slos: [slos$, (value) => slos$.next(value)],
+          showAllGroupByInstances: [
+            showAllGroupByInstances$,
+            (value) => showAllGroupByInstances$.next(value),
+          ],
+          ...titleComparators,
+        }
+      );
+
+      const fetchSubscription = fetch$(api)
+        .pipe()
+        .subscribe((next) => {
+          reload$.next(next);
+        });
+
+      return {
+        api,
+        Component: () => {
+          const [slos, showAllGroupByInstances] = useBatchedPublishingSubjects(
+            slos$,
+            showAllGroupByInstances$
+          );
+          const fetchContext = useFetchContext(api);
+          const I18nContext = deps.i18n.Context;
+
+          useEffect(() => {
+            return () => {
+              fetchSubscription.unsubscribe();
+            };
+          }, []);
+          return (
+            <I18nContext>
+              <KibanaContextProvider
+                services={{
+                  ...deps,
+                  storage: new Storage(localStorage),
```

Comment:
thi should use `SloEmbeddableContext`

Reply:
@shahzad31 I was comparing the newly introduced [context file ](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/slo/public/embeddable/slo/common/slo_embeddable_context.tsx) with the [current file ](https://github.com/mgiota/kibana/blob/refactor_alerts_embeddable/x-pack/plugins/observability_solution/slo/public/embeddable/slo/alerts/slo_alerts_embeddable_factory.tsx) and since there are a few differences, I want to avoid doing this change here. I want to keep this PR only for the refactoring to the new Embeddable system. This change would require more testing to the Overview embeddable as well.